app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala

```diff
@@ -37,21 +44,42 @@ class GoodsShipmentBuilder @Inject()(
     exportsCacheModel.natureOfTransaction.foreach(
       natureOfTransaction => goodsShipmentNatureOfTransactionBuilder.buildThenAdd(natureOfTransaction, goodsShipment)
     )
+
     exportsCacheModel.parties.consigneeDetails
       .foreach(consigneeDetails => consigneeBuilder.buildThenAdd(consigneeDetails, goodsShipment))
-    ConsignmentBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
-    DestinationBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
-    ExportCountryBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
-    UCRBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
-    WarehouseBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
-    PreviousDocumentsBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
+
+    consignmentBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
+
+    exportsCacheModel.locations.destinationCountries
+      .foreach(destinationCountries => {
+        destinationBuilder.buildThenAdd(destinationCountries, goodsShipment)
+        exportCountryBuilder.buildThenAdd(destinationCountries, goodsShipment)
+      })
+
+    exportsCacheModel.consignmentReferences.foreach(
+      consignmentReferences => ucrBuilder.buildThenAdd(consignmentReferences, goodsShipment)
+    )
+
+    exportsCacheModel.locations.warehouseIdentification
+      .foreach(warehouseIdentification => warehouseBuilder.buildThenAdd(warehouseIdentification, goodsShipment))
+
+    exportsCacheModel.previousDocuments.foreach(
+      previousDocuments => previousDocumentsBuilder.buildThenAdd(previousDocuments, goodsShipment)
+    )
+
+    exportsCacheModel.parties.declarationAdditionalActorsData.foreach { declarationAdditionalActorsData =>
+      {
```

Comment:
Would you like to explain purpose of this scope ?

Reply:
I think what he means is you probably don't need the `{` and `}`

Reply:
It reads better for me as you can see the scope as you have nested foreach. But don't have strong feelings about it either way.